Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml

```diff
@@ -0,0 +1,69 @@
+id: f4a28082-2808-4783-9736-33c1ae117475
+name: High-Risk Cross-Cloud User impersonation 
+description: |
+  'This detection focuses on dentifying high-risk cross-cloud activities and sign-in anomalies that may indicate potential security threats. The query starts by analyzing Azure AD SigninLogs to pinpoint instances where specific applications, risk levels, and result types align. It then correlates this information with relevant AWS CloudTrail events to identify activities across Azure and AWS environments.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1134
+  - T1078.002
+  - T1078.004
+query: |
+  // Retrieve Azure AD SigninLogs within the last day
+  SigninLogs 
+  | where TimeGenerated >= ago(1d)
+  // Filter for specific AppDisplayNames, ResultType, and Risk Levels
+  | where AppDisplayName in ("Azure Portal", "ADFS Trust", "Microsoft Azure PowerShell")
+      and ResultType == 0

```

Comment:
Given we are joining on just IP why wouldn't we also want to capture risky failed logons?

Reply:
objective of this query is to detect the successful High risk SignIn and the find suspicious activity in AWS, for fil logon we can create another detection

Reply:
I get that but wouldn't a failed high risk login attempt performing some action in AWS also be worth detecting? If I am seeing Azure blocking and IP due to some risk factor I would definitely want to know if that IP then starts doing stuff in AWS. I would even argue that this is more valuable that a high risk successful sign in from Azure as the success of that sign in suggests that the high risk tag may have been mitigated by the user.

Reply:
Got your point, I have removed the ResultType== 0 condition and added description of event as below.
| extend Result = iff(ResultType == 0, "Successful Signin", "Failed Signin")